### ISSUE TITLE
Can't create Augmented Images dynamically when doing an async call

### ISSUE THREAD

**Issue by tjsearsz** [SceneView/sceneview-android#291]
Hello,

First and foremost, thank you for maintaining and improving this incredible library. 

At the moment, I am trying to generate Augmented Images by fetching from a server the modelURL. I see that I am able to fetch the information successfully, but when I am creating the node, nothing happens. Here is the following code (this is inside a Fragment):

```
override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

sceneView = view.findViewById(R.id.sceneView)

lifecycleScope.launchWhenCreated {

            var result =  ApolloAugmentedImageClient().getAugmentedImages()
            
            sceneView.addChild(AugmentedImageNode(
                sceneView.engine,
                imageName = "rabbit",
                bitmap = requireContext().assets.open("augmentedimages/image.png")
                    .use(BitmapFactory::decodeStream)
            ).apply { loadModelGlbAsync(glbFileLocation = "https://myurl.com") })
        }
}
```

As you can see, I tried using static content (asset folder) to see what was happening and check if maybe I wasn't getting the values correctly from the server. If I remove the `result` line, the code works. But with the call `(getAugmentedImages)` it doesn't work. For some reason, it cannot render/create the images with an asynchronous call before the `addChild` even if the values are in the `asset` folder. 

I already tried placing the sceneView inside the `launchWhenCreated`, used `GlocalScope` and `async`, but no success so far. I already checked whether the images were saved in the sceneView, and I see that sceneView already has them in the array of children (`List<Node>`). 

For example, the following code does not work:

```
lifecycleScope.launchWhenCreated {
            val result = async { ApolloAugmentedImageClient().getAugmentedImages() }

            result.await()

            sceneView.addChild(
                AugmentedImageNode(
                    sceneView.engine,
                    imageName = "rabbit",
                    bitmap = requireContext().assets.open("augmentedimages/image.png")
                        .use(BitmapFactory::decodeStream)
                ).loadModelGlbAsync(glbFileLocation = "https://myurl.com"))
}

```

However, the following code works:

```
lifecycleScope.launchWhenCreated {
            val result = async { ApolloAugmentedImageClient().getAugmentedImages() }

            // result.await()      <---- Notice the commented line

            sceneView.addChild(
                AugmentedImageNode(
                    sceneView.engine,
                    imageName = "rabbit",
                    bitmap = requireContext().assets.open("augmentedimages/image.png")
                        .use(BitmapFactory::decodeStream)
                ).loadModelGlbAsync(glbFileLocation = "https://myurl.com"))
}

```
My guess is that it should be related to something regarding `loadModelGlbAsync`, but I am not sure unfortunately.

The idea is that `result` is an object that contains the URL of the model `modelURL`, and then simply add that URL as parameter for the `glbFileLocation` of the `loadModelGlbAsync.`

I would appreciate if anyone could give me a small support on this.

**Comment by grassydragon** [comment]
Hi!
Could you provide a reference for this library to learn more about it?
```kotlin
ApolloAugmentedImageClient().getAugmentedImages()
```

**Comment by tjsearsz** [comment]
Hello @grassydragon !

First and foremost, thank you for reviewing this inquiry I made.

This is a class I created for performing a simple GraphQL call using Apollo Kotlin Library. However, since I was debugging to see whether the issue could potentially be related to this library. I made the following test:

```
override fun getAugmentedImages(): List<MyAugmentedImage> {
        delay(5000L)      <--- Notice this simple Delay
        return listOf(
            MyAugmentedImage(
                modelURL = "https://myurl.com",
                imageURL = "https://myurl.com"
            ),
            MyAugmentedImage(
                modelURL = "https://myurl.com",
                imageURL = "https://myurl.com"
            )
        )
    }
```

As you can see, this is just simple suspended method that has a 5000L delay and returns hard-coded values.

 `MyAugmentedImage` is just a simple data class

`data class MyAugmentedImage(val imageURL: String, val modelURL: String`

And this implementation does not work. As I mentioned in my initial comment, if I try to `await` this function, or even if I just place this method before the `sceneView.addChild` in a coroutine the augmented reality will not work for some reason. 

In other words, it looks like placing a suspended function before the `sceneView.addChild` makes the library to not work even if I am using the image and model from the `assets` folder.

To reproduce this error, we just simply create a simple android kotlin application, create a suspended function with a small delay that returns hardcoded values (e.g. the one that I showed before) and place this function before the addChild.

If you need any other information from my side, please kindly let me know.

Sincerely
